crates/bevy_render/src/render_resource/storage_buffer.rs

```diff
@@ -214,14 +223,18 @@ impl<T: ShaderType + WriteInto> DynamicStorageBuffer<T> {
         self.label.as_deref()
     }
 
+    pub fn set_usage(&mut self, usage: BufferUsages) {
+        self.buffer_usage.set(usage, true);
+    }
+
     #[inline]
     pub fn write_buffer(&mut self, device: &RenderDevice, queue: &RenderQueue) {
         let size = self.scratch.as_ref().len();
 
         if self.capacity < size || self.label_changed {
```

Comment:
We probably want to update the buffer the next time it is written to, after the usages have changed.
We could rename the `label_changed` flag to `changed` and update it if the usage has changed as well.